packages/statemanager/src/stateManager.ts

```diff
@@ -335,21 +344,17 @@ export class DefaultStateManager implements EVMStateManagerInterface {
    * @param value - The value of the `code`
    */
   async putContractCode(address: Address, value: Uint8Array): Promise<void> {
+    const currentCode = await this.getContractCode(address)
+    this._codeCache2?.put(address, value, currentCode)
```

Comment:
Hmm, interesting. It is not possible that `currentCode` is non-empty and then we try to put a different `value` there (it is not possible to overwrite contracts). Can we optimize further here? :thinking:

Reply:
Yes, I simplified it a bit since all we care about is if the account has a contract or not, which is used for reverting code.